arbos/programs/api.go

```diff
@@ -314,14 +320,20 @@ func newApiClosures(
 			input = []byte{}
 			return data
 		}
+		takeGasLeft := func() uint64 {
+			defer func() {
+				recordGasOpcode()
+			}()
+			return takeU64()
+		}
 
 		switch req {
 		case GetBytes32:
 			key := takeHash()
 			out, cost := getBytes32(key)
 			return out[:], nil, cost
 		case SetTrieSlots:
-			gasLeft := takeU64()
+			gasLeft := takeGasLeft()
```

Comment:
I don't think we want these..
This is information passed from stylus to geth to understand if there is enough gas to complete the request - but GAS opcode should only be added to the trace where the smart contract checked how much gas is left.
The smart contract does that by calling one of the functions evm_gas_left or evm_ink_left.

I think (not sure), that when these functions are called while tracing is enabled it creates a request that will arrive here as CapturHostIo.
If I am correct you should check the hostio name in captureHostio, and if it's evm_gas_left or evm_ink_left - mark that in the trace as a gall to vm.GAS + vm.POP

Reply:
My bad, I was registering every `self.gas_left()` invocation on the rust side instead of hostio. Fixed it